airflow/executors/kubernetes_executor.py

```diff
@@ -681,6 +714,78 @@ def _change_state(self,
                 self.log.debug('Could not find key: %s', str(key))
         self.event_buffer[key] = state, None
 
+    def try_adopt_task_instances(self, tis: List[TaskInstance]) -> List[TaskInstance]:
+        tis_to_flush = [ti for ti in tis if not ti.external_executor_id]
+        worker_uuids = [ti.external_executor_id for ti in tis]
+        pod_ids = {
+            create_pod_id(dag_id=ti.dag_id, task_id=ti.task_id): ti
+            for ti in tis if ti.external_executor_id
+        }
```

Comment:
This can be refactored similarly to https://github.com/apache/airflow/blob/7644c3708223afef1d9e80cdd7d991bf9cddfa8e/airflow/executors/celery_executor.py#L350-L357

similar to https://github.com/apache/airflow/pull/10949#discussion_r489508144

Reply:
@kaxil I was thinking... we might not need external_uuids for this. Even if it was launched by a "legacy" k8sExecutor. We're just changing the label and monitoring. WDYT?